### PR TITLE
Glossary API 

### PR DESCRIPTION
**Pull request by GoddessEyes**
* add `update_glossary` query
* add required name for Glossary
* fix pydantic deprecation`s
* add /glossary/{glossary_id}/records/ endpoint
* Rename GlossaryDoc to Glossary
* Add glossary_to_document M2M red

### CODE REVIEW

File: backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# pylint: disable=E1101
+
+# revision identifiers, used by Alembic.
+revision: str = "871e5285a2a3"
+down_revision: Union[str, None] = "bc813d99ccfb"
```
Comment (ArchiDevil) [inline backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py:19]:
```suggestion
down_revision: Union[str, None] = "96bf2e203c03"
```
As my PR was merged.

File: backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
```
[the remark this answers is not in the document]
Comment (GoddessEyes) [reply backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py]: Fixed

File: backend/app/routers/glossary.py [hunk 3]
```diff
@@ -64,6 +68,46 @@ def retrieve_glossary_doc(glossary_doc_id: int, db: Session = Depends(get_db)):
...
+    if glossary_doc_response:
+        return glossary_doc_response
+    raise HTTPException(
+        status_code=status.HTTP_404_NOT_FOUND,
+        detail=f"Glossary document id:{glossary_doc_id}, not found",
+    )
+
+
+@router.get(
+    "/records",
```
[the remark this answers is not in the document]
Comment (GoddessEyes) [reply backend/app/routers/glossary.py]: Fixed

File: backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py (hunk 1 above)
Comment (GoddessEyes) [reply backend/alembic/versions/871e5285a2a3_add_name_for_glossary_doc.py]: Fixed

**Review by ArchiDevil — APPROVED** [review]

**Pull request merged**